AndroidSDK/src/com/leanplum/internal/LeanplumManifestHelper.java

```diff
@@ -78,7 +78,10 @@
   /**
    * Gets application components from AndroidManifest.xml file.
    */
-  private static void parseManifestNodeChildren() {
+  private static synchronized void parseManifestNodeChildren() {
+    if (manifestData != null) {
```

Comment:
Because if manifest is already parsed there is no reason to do it again. It will not change until application is reinstalled.
And it is not immediate operation to parse it because data needs to be pulled from .zip archive